Update TailwindCSS to v4 and added sample config for TALL Stack compatibility.

omimouni opened the pull request:
Description:

Laravel 12 ships with TailwindCSS 4 by default, which causes compatibility issues when installing and using the TALL stack. This PR updates the necessary packages and adds a sample TailwindCSS 4 configuration file, ensuring it follows the same approach as the forms and typography plugins.

omimouni committed:
Fix: Update packages for Tailwind CSS 4, clean up package list by removing @tailwindcss/vite and sass, and set PostCSS as default. Added @tailwindcss/nesting for better nesting support and fixed PostCSS config.

danharrin commented on stubs/default/resources/sass/tailwind.css:
Can we keep the Tailwind in the `app.scss` instead? Also maybe lets use this opportunity to rename `app.scss` to `app.css` since many important sass features like nesting are supported by PostCSS.

At the same time, rename `sass` to `css`

omimouni replied:
Done! I did test the package locally on a fresh built as well it seems to work fine.

n355 commented:
As per [Tailwind CSS's official documentation](https://tailwindcss.com/docs/installation/using-postcss), when using Tailwind CSS with PostCSS, only @tailwindcss/postcss and postcss are required. 
Autoprefixer is not needed as Tailwind now handles prefixing internally.
It might be a good idea to include the `@tailwindcss/nesting` package to replace Sass nesting functionality.

danharrin commented on src/TallPreset.php:
```diff
@@ -15,7 +15,9 @@ class TallPreset extends Preset
         'sass-loader' => '^8.0',
-        'tailwindcss' => '^3.4',
+        'tailwindcss' => '^4.0',
+        '@tailwindcss/postcss' => '^4.0',
```
I don't think we need this, right?

n355 commented:
@danharrin I tested a Laravel 12 project with `postcss` and `@tailwindcss/postcss`. Tailwind works with Vite without the `@tailwindcss/vite` plugin, FYI.

Note that Tailwind 4's default nesting doesn't support the `&` parent reference.

danharrin commented:
Look at the Laravel 12 skeleton, there is no Postcss at all

n355 commented:
I am well aware that Laravel 12 comes without a PostCSS config, but since I saw in this PR that you were planning to keep it, I was speaking in context—meaning I uninstalled the default Vite plugin to switch to a PostCSS-only setups. 
Hope this clarifies my point.

danharrin commented:
I think we should use this PR to slim down the changes required to use the preset, so that means completely removing PostCSS, sass, etc. It will be clearer to people familiar with Laravel 12. We can drop nesting support out of the box if we don't use it in any of our code either.

omimouni commented:
in the last commit I cleaned up package list by removing @tailwindcss/vite and Sass, and set PostCSS as default. Added @tailwindcss/nesting for better nesting and fixed PostCSS config. I tested it and it works well and blazing fast.

omimouni commented:
> I think we should use this PR to slim down the changes required to use the preset, so that means completely removing PostCSS, sass, etc. It will be clearer to people familiar with Laravel 12. We can drop nesting support out of the box if we don't use it in any of our code either.

I believe keeping PostCSS and nesting functionality would facilitate an easier migration to the newer version of Laravel and the TALL stack while maintaining backward compatibility. Speaking from experience, I've used this package in most of my Laravel projects, and these features have been quite beneficial.

danharrin commented:
This package is for new projects only and not upgrading existing projects, so please remove PostCSS and nesting to keep the installation minimal @omimouni

n355 commented:
@omimouni I just ran into build issues with the @tailwindcss/nesting plugin. Turns out it’s not maintained and doesn’t work properly. I’d recommend sticking to the official method with the Vite plugin instead, as @danharrin suggests.

omimouni committed:
fix: used tailwindcss with vite, and removing PostCSS

omimouni commented:
@n355 @danharrin I kept only the minimal packages for Tailwind CSS to work and configured Vite to match Laravel 12's default setup. I think this is all set now

danharrin commented on src/TallPreset.php:
Lodash is no longer installed by default so we can remove that
Tailwind and Tailwind Vite don't need to be added since they are already in the skeleton

danharrin commented on stubs/default/vite.config.js:
This file can be removed since its the same as the default `vite.config.js` in the app

omimouni committed:
removed vite config as it’s the same as the default, and removed packages that comes with laravel 12

danharrin reviewed (changes requested):
The tests are failing because of the outdated CI workflow, can you take a look please?

omimouni committed:
fix: upgraded actions/cache version in ci/cd

omimouni commented:
> The tests are failing because of the outdated CI workflow, can you take a look please?

Yes seems like actions/cache v1 was deprecated, I did change it to version v4!

danharrin commented:
Thank you
